CMakeLists.txt

```diff
@@ -0,0 +1,298 @@
+cmake_minimum_required(VERSION 2.6)
+
+project("openscap")
+set(OPENSCAP_VERSION_MAJOR "1")
+set(OPENSCAP_VERSION_MINOR "3")
+set(OPENSCAP_VERSION_PATCH "0")
+set(OPENSCAP_VERSION "${OPENSCAP_VERSION_MAJOR}.${OPENSCAP_VERSION_MINOR}.${OPENSCAP_VERSION_PATCH}")
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
+
+if(CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
+	set(NDEBUG 1)
+endif()
+
+# ---------- INCLUDES CMAKE MODULES
+
+include(GNUInstallDirs)
+include(FindPkgConfig)
+include(CheckLibraryExists)
+include(CheckFunctionExists)
+include(CheckIncludeFile)
+include(CheckCSourceCompiles)
+include(CPack)
+
+# ---------- DEPENDENCIES
+
+# ACL
+find_package(ACL)
+if(ACL_FOUND)
+	check_library_exists(acl acl_extended_file "" HAVE_ACL_EXTENDED_FILE)
+	check_include_file(acl/libacl.h HAVE_ACL_LIBACL_H)
+	check_include_file(sys/acl.h HAVE_SYS_ACL_H)
+endif()
+
+# APTPKG
+find_package(AptPkg)
+
+# BLKID
+find_package(Blkid)
+if(BLKID_FOUND)
+	check_library_exists(blkid blkid_get_tag_value "" HAVE_BLKID_GET_TAG_VALUE)
+endif()
+
+# CAP
+find_package(Cap)
+if(CAP_FOUND)
+	check_library_exists(cap cap_get_pid "" HAVE_CAP_GET_PID)
+endif()
+
+# CURL
+find_package(CURL)
+
+# DBUS
+find_package(DBus)
+
+# DOXYGEN
+find_package(Doxygen)
+
+# GCONF2
+find_package(GConf)
+
+# LDAP
+find_package(Ldap)
+
+# OpenDbx
+find_package(OpenDbx)
+
+# PCRE
+find_package(PCRE REQUIRED)
+
+# PERL
+find_package(PerlLibs)
+
+# POPT
+find_package(Popt)
+
+# PROCPS
+find_package(Procps)
+if(PROCPS_FOUND)
+	check_library_exists(procps dev_to_tty "" HAVE_DEV_TO_TTY)
+	check_include_file(proc/devname.h HAVE_PROC_DEVNAME_H)
+endif()
+
+# PYTHON
+find_package(PythonLibs 2 EXACT)
+find_package(PythonInterp)
+
+# RPM
+find_package(RPM)
+if(RPM_FOUND)
+	check_library_exists(rpm rpmReadConfigFiles "" HAVE_RPMREADCONFIGFILES)
+	check_library_exists(rpm headerFormat "" HAVE_HEADERFORMAT)
+	check_library_exists(rpmio rpmFreeCrypto "" HAVE_RPMFREECRYPTO)
+	check_library_exists(rpm rpmFreeFilesystems "" HAVE_RPMFREEFILESYSTEMS)
+	set(HAVE_RPMVERCMP 1)
+endif()
+
+# SELINUX
+find_package(SELinux)
+
+# SWIG
+find_package(SWIG)
+
+# XML2
+find_package(LibXml2 REQUIRED)
+
+# XSLT
+find_package(LibXslt REQUIRED)
+
+# BZip2
+find_package(BZip2)
+
+# PThread
+set(CMAKE_THREAD_PREFER_PTHREAD)
+set(THREADS_PREFER_PTHREAD_FLAG)
+find_package(Threads)
+check_library_exists(pthread pthread_timedjoin_np "" HAVE_PTHREAD_TIMEDJOIN_NP)
+check_library_exists(pthread pthread_setname_np "" HAVE_PTHREAD_SETNAME_NP)
+check_library_exists(pthread pthread_getname_np "" HAVE_PTHREAD_GETNAME_NP)
+
+# WITH_CRYPTO
+set(WITH_CRYPTO "gcrypt" CACHE STRING "gcrypt|nss3")
+if(NOT (${WITH_CRYPTO} EQUAL "nss3"))
+	# gcrypt
+	find_package(GCrypt)
+else()
+	# nss3
+	find_package(NSS)
+endif()
```

Comment:
I don't understand this logic at all. Shouldn't you instead look for both and then set WITH_CRYPTO accordingly?

Reply:
@mpreisler I think that's because you can choose which of the two libraries you want to use.

Reply:
This is something we should change in the future but for now we can keep this weird convention. Usually projects will look for the deps and then give you the choice based on what they find. So for example if you only have NSS but don't have GCrypt it won't even give you the option to choose GCrypt.